pkgs/development/compilers/llvm/common/default.nix

```diff
@@ -990,6 +1005,8 @@ let
         stdenv =
           if stdenv.hostPlatform.isDarwin && stdenv.hostPlatform == stdenv.buildPlatform then
             stdenv
+          else if stdenv.targetPlatform.isAndroid then
+            overrideCC stdenv buildLlvmTools.clangLibcNoRt
           else
             # TODO: make this branch unconditional next rebuild
             overrideCC stdenv buildLlvmTools.clangNoLibcNoRt;
```

Comment:
I guess we're not making this unconditional any more then?

Why is Android special here?  What's the effect on non-prebuilt Android toolchains?

Reply:
compiler-rt for android needs libc in order to build, which is what this does.

The non-prebuilt android toolchain(s) fail to build for another reason...

Reply:
Could that be explained in a comment? Ideally with reference to where this distinction is made in the compiler-rt source code?

Reply:
Sure. I'll make that change tomorrow morning.

Reply:
Added a comment.